script.js

```diff
@@ -0,0 +1,165 @@
+const toDoForm = document.querySelector("form");
+const toDoInput = toDoForm.querySelector(".input-text");
+const toDoButton = toDoForm.querySelector(".input-button");
+
+const TODO_DB = "todo";
+let toDoList = [];
+
+// localStorage 모델: 웹이 로드 될 때 기존 투두리스트를 가져오기 위함
+const localStorageModel = {
+    // localStorage에 저장
+    save: () => {
+        localStorage.setItem(TODO_DB, JSON.stringify(toDoList));
+    },
+
+    // localStorage에서 로드
+    load: () => {
+        const loaded = localStorage.getItem(TODO_DB);
+        if (loaded !== null) {
+            const parsed = JSON.parse(loaded);
+            toDoList = parsed;
+        }
+    }
+}
+
+// List 모델
+const list = {
+    // 개수 계산
+    getCount: () => {
+        let pending = 0;
+        toDoList.forEach((item) => {
+            if (!item.done) {
+                pending++;
+            }
+        })
+        return {
+            pending: pending,
+            done: toDoList.length - pending
+        }
+    },
+
+    // list 대기중 완료 토글
+    updateList: (value, isDone) => {
+        const i = toDoList.findIndex((item) => {
+            return item.text == value;
+        })
+        toDoList[i].done = toDoList[i].done == isDone ? !isDone : isDone;
+    },
+
+    // toDoList 배열에 추가  -> submit 할 때
+    addToList: (value) => {
+        let item = {
+            "id": toDoList.length + 1,
+            "text": value,
+            "done": false
+        }
+        toDoList.push(item);
+    },
+
+    // toDoList 배열에서 제거 -> 휴지통
+    removeFromList: (value) => {
+        const i = toDoList.findIndex((item) => {
+            return item.text == value;
+        })
+        toDoList.splice(i, 1);
+    }
+}
+
+// Paint 모델: 화면에 그리고 빼는 것과 관련
+const paint = {
+    // 새로고침 화면 그리기
+    loadData: () => {
+        localStorageModel.load();
+        
+        paint.updateList();
+        paint.listCount();
+    },
+
+    // 리스트 개수 그리기
+    listCount: () => {
+        document.querySelector(".count-pending").innerHTML = list.getCount().pending;
+        document.querySelector(".count-done").innerHTML = list.getCount().done;
+    },
+
+    // toDoList 다시 그리기
+    updateList: () => {
+        const ul = document.querySelectorAll("ul");
```

Comment:
헉. ul이라고만 하면
1. ul이 document의 ul 중 어떤 엘레멘트를 가르키는지
2. ul이 ul태그를 가진 엘레멘트의 리스트인지 아니면 한개의 엘레멘트인지 모르죠?

게다가 ul[0]라고 고정된 인덱스로 접근해버리면 또다른 ul 태그가 문서에 추가됐을때 코드가 다 무너지게 됩니다 ㅠㅠ 이럴때는 element id를 이용해서 ul element를 특정해주세요!

Reply:
현재 ul이 2개 있어서 `querySelectorAll()`을 통해 ul을 전부 가져온 후 그 리스트 내의 모든 값을 지우려고 했어요..!
element id를 이용하면 위에서부터 발견되는 id를 집어내와 대기중과 완료 두개 모두의 ul DOM을 못 가져오는데 이럴경우 어떻게 id로 2개의 값을 다 가져올 수 있을까요?

Reply:
뭔가 코드 상에서 'ul 내부를 다 지운다.' 라는 의미보다는 '완료리스트, 대기리스트를 비운다'라는 의미로 작성하는게 좋을 것 같다는 생각이었어요 ㅎㅎ(가독성 높이기) 이렇게 해볼 수 있겠죠?

```html
<h3 class="title title-pending">
대기중 (<span class="count count-pending">0</span>)
</h3>
<ul id="list-pending"></ul>
<h3 class="title title-done">
완료됨 (<span class="count count-done">0</span>)
</h3>
<ul id="list-done"></ul>
```

```suggestion
        const pendingTodoList = document.getElementById(".list-pending");
        const doneTodoList = document.getElementById(".list-done");
        pendingTodoList.innerHTML = '';
        doneTodoList.innerHTML = "";
```